### PR TITLE
Gpx track appearance

### PR DESCRIPTION
**Pull request by Chumva**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/helpers/ImportHelper.java [hunk 1]
```diff
@@ -1018,7 +1019,18 @@ private String saveImport(final GPXFile gpxFile, final String fileName, final bo
 					File file = new File(gpxFile.path);
 					if (!destinationExists) {
 						GPXDatabase.GpxDataItem item = new GPXDatabase.GpxDataItem(file, gpxFile.getColor(0));
+						item.setWidth(gpxFile.getWidth(null));
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/helpers/ImportHelper.java:1022]: Introduce new constructor **public GpxDataItem(File file, GPXFile gpx)** and drop all these setters.

File: OsmAnd/src/net/osmand/plus/views/GPXLayer.java [hunk 2]
```diff
@@ -213,36 +209,36 @@ public void onPrepareBufferImage(Canvas canvas, RotatedTileBox tileBox, DrawSett
...
-			currentTrackColorPref.get(), currentTrackWidthPref.get());
-		if (hsh != cachedHash) {
-			cachedHash = hsh;
+		boolean nightMode = drawSettings != null && drawSettings.isNightMode();
+		int hash = calculateHash(rrs, cachedTrackWidth, routePoints, nightMode, tileBox.getMapDensity(), tileBox.getZoom(),
+				currentTrackColorPref.get(), currentTrackWidthPref.get());
+		if (hash != cachedHash) {
+			cachedHash = hash;
 			cachedColor = ContextCompat.getColor(view.getApplication(), R.color.gpx_track);
+			defaultTrackWidth = 7 * view.getDensity();
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/views/GPXLayer.java:228]: 7?

File: OsmAnd/src/net/osmand/plus/views/GPXLayer.java [hunk 3]
```diff
@@ -262,32 +258,60 @@ private int updatePaints(int color, boolean routePoints, boolean currentTrack, D
...
 		}
 		paint.setColor(color == 0 ? cachedColor : color);
+		Float strikeWidth = cachedTrackWidth.get(width);
+		if (strikeWidth != null) {
+			paint.setStrokeWidth(strikeWidth);
+		}
 		return cachedColor;
 	}
 
+	private void searchTrackWidth(String widthKey, RenderingRulesStorage rrs, RenderingRuleSearchRequest req, RenderingContext rc) {
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/views/GPXLayer.java:287]: searchTrackWidth > fetchTrackWidth or acquireTrackWidth

File: OsmAnd/src/net/osmand/plus/views/GPXLayer.java (hunk 2 above)
Comment (Chumva) [reply OsmAnd/src/net/osmand/plus/views/GPXLayer.java:228]: It was moved from here - https://github.com/osmandapp/Osmand/pull/9330/files#diff-3259c804faea84b62ab64ab1e0f87b01L277 for cases where track width parameter wasn't acquired from renderer or custom width has an invalid value.

**Pull request merged**